src/v/resource_mgmt/storage.cc

```diff
@@ -110,13 +122,59 @@ ss::future<> disk_space_manager::run_loop() {
             continue;
         }
 
-        if (!_log_storage_target_size().has_value()) {
+        if (_target_size == 0) {
             _local_monitor->local().set_log_data_state(std::nullopt);
             continue;
         }
 
-        co_await manage_data_disk(_log_storage_target_size().value());
+        co_await manage_data_disk(_target_size);
+    }
+}
+
+void disk_space_manager::update_target_size() {
+    // amount of data disk reserved for non-redpanda use
+    const uint64_t reservation_size = _data_disk_size
+                                      * (_disk_reservation_percent() / 100.0);
+
+    // unreserved data disk space
+    const auto usable_size = _data_disk_size - reservation_size;
+
+    // percent-based target size
+    const uint64_t target_size_pct
+      = usable_size
+        * (_retention_target_capacity_percent().value_or(0) / 100.0);
+
+    // bytes-based target size
+    const uint64_t target_size_bytes
+      = _retention_target_capacity_bytes().value_or(0);
```

Comment:
Should this be capped to the usable size?

Reply:
fixed